src/runtime_src/core/edge/drm/zocl/zocl_sysfs.c

```diff
@@ -133,6 +133,38 @@ static ssize_t zocl_get_memstat(struct device *dev, char *buf, bool raw)
 	return size;
 }
 
+static ssize_t read_aie_metadata(struct file *filp, struct kobject *kobj,
+		struct bin_attribute *attr, char *buf, loff_t off, size_t count)
+{
+	struct drm_zocl_dev *zdev;
+	size_t size;
+	u32 nread = 0;
+
+	zdev = dev_get_drvdata(container_of(kobj, struct device, kobj));
+	if (!zdev)
+		return 0;
+
+	read_lock(&zdev->attr_rwlock);
+
+	size = zdev->aie_data.size;
+
+	if (off >= size) {
+		read_unlock(&zdev->attr_rwlock);
+		return 0;
+	}
+
+	if (count < size - off)
```

Comment:
This code looks "important", could you add a comment what is happening here.

Reply:
done